### PR TITLE
Create docs/index.rst file

### PR DESCRIPTION
**Pull request by RobertKoikai**
Depends on the-deep/server 

## Changes

When building documentation using Read the Docs, it looks for the index.rst file as the starting point. It parses this file and generates the documentation output accordingly. The index.rst file typically contains a hierarchical structure of headings and subheadings, which represent the organisation and navigation of your documentation.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [x] permission checks (tests here too)
- [x] translations

### CODE REVIEW

File: docs/source/Options/APIReference.rst
Comment (thenav56) [inline docs/source/Options/APIReference.rst:1]: We have both graphql and REST in deep. Most of the platform uses graphql but there are some parts where we still use REST which we are trying to migrate from as well.

For graphql: The schema is synced here https://github.com/the-deep/server/blob/develop/schema.graphql
For REST: It is not maintained using git but can be retrieved using API. For prod: https://prod-api.thedeep.io/api-docs/?format=openapi

Maybe we can look into tools that can take this schema as input and add content to this file.

File: docs/source/Options/Changelog.rst
Comment (thenav56) [inline docs/source/Options/Changelog.rst:1]: Right now we have this wiki where we add changes logs.
https://github.com/the-deep/deeper/wiki
Maybe we can point to there.

cc: @AdityaKhatri

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good.

Let's clean up the build/ directory using .gitignore.

**Comment by thenav56** [comment]
Hi @RobertKoikai
Let's also add the poetry snippet documentation we discussed on https://github.com/the-deep/server/pull/1325
```markdown
## Add new package

- In the server directory
  - Add package in pyproject.yml file
  - Run `poetry lock --no-update`. This will update poetry.lock
- In the deeper directory
  - docker compose up --build
```

File: .gitignore [hunk 1]
```diff
@@ -47,3 +47,5 @@ schema-latest.graphql
 
 ci-share/
 rest-media-temp/
+
+docs/build/
```
Comment (thenav56) [inline .gitignore:51]: Let's remove the files from git as well.

**Review by thenav56 — CHANGES_REQUESTED** [review]

**Comment by thenav56** [comment]
@RobertKoikai @AdityaKhatri 

The changes look good. But Looking at the changes here seems like we need to add this to the the-deep repo instead of the server.
As all of the documentation relates to overall deep instead of just server.

Let me know if you are planning to also add auto-generated code documentation as well.

**Comment by RobertKoikai** [comment]
> @RobertKoikai @AdityaKhatri
> 
> The changes look good. But Looking at the changes here seems like we need to add this to the the-deep repo instead of the server. As all of the documentation relates to overall deep instead of just server.
> 
> Let me know if you are planning to also add auto-generated code documentation as well.

Hi @thenav56 ,
Maybe @AdityaKhatri can advice on that.
In regards to autogenerated code documentation yes that is something that can also be done in the second phase of the implementation of the documentation.

cc : @nayid-DFS

**Comment by thenav56** [comment]
> > @RobertKoikai @AdityaKhatri
> > The changes look good. But Looking at the changes here seems like we need to add this to the the-deep repo instead of the server. As all of the documentation relates to overall deep instead of just server.
> > Let me know if you are planning to also add auto-generated code documentation as well.
> 
> Hi @thenav56 , Maybe @AdityaKhatri can advice on that. In regards to autogenerated code documentation yes that is something that can also be done in the second phase of the implementation of the documentation.
> 
> cc : @nayid-DFS

Let's move this if we are planning to do autogenerated code documentation later. We can later introduce sub-modules and then use the server to generate from the deeper repo.

**Comment by RobertKoikai** [comment]
@thenav56 ,
Seems like I do not
Screenshot 2023-06-22 at 2 45 46 PM
 have the correct permissions for the https://github.com/the-deep/deeper repository.See screenshot below ;

**Comment by thenav56** [comment]
> @thenav56 , Seems like I do not Screenshot 2023-06-22 at 2 45 46 PM have the correct permissions for the https://github.com/the-deep/deeper repository.See screenshot below ;

Sure, this is resolved.
Let's move this conversation to the-deep/deeper :D

**Pull request closed without merging**